### ISSUE TITLE
Test duration sorts with duration string rather than the actual runtime

### ISSUE THREAD

**Issue by anthony-chu** [DevExpress/testcafe#7485]
### What is your Scenario?

When sorting by duration, test duration should be sorted by the runtime indicated by the string rather than the string value of the duration itself.

### What is the Current behavior?

Duration sorting sorts the test durations by the human friendly duration string rather than the actual runtime (see screenshot)

### What is the Expected behavior?

Test results should be sorted by the total runtime of each test (combining all the valid time units of hours, minutes, seconds, etc...)

### What is your public website URL? (or attach your complete example)

n/a

### What is your TestCafe test code?

n/a

### Screenshots

![image](https://user-images.githubusercontent.com/8292593/216142789-37997044-e118-4b82-a332-b276236ca2ae.png)

### Steps to Reproduce

1. Run some tests
    - Ensure tests have different run times (some in seconds, some in minutes)
2. View the HTML report of the test suite
3. Sort the test results by **Duration**

### TestCafe version

1.20.1

### Command-line arguments

n/a

**Comment by Artem-Babich** [comment]
Hello,

This reporter isn't supported by the TestCafe team. Please leave all future feedback in the owner's repository. However, I saw the reporter's code and created a PR with the fix: ([fix: sort duration column by value instead of text](https://github.com/picuscreative/testcafe-reporter-html/pull/56)). You will be able to use it after the owner of the repository merges the PR.

**Comment by Artem-Babich** [comment]
Hi @anthony-chu
I see that the [testcafe-reporter-html](https://github.com/picuscreative/testcafe-reporter-html) repo owner didn't merge [my PR with the fix](https://github.com/picuscreative/testcafe-reporter-html/pull/56). I would like to follow up on this issue and share a solution that will help you to fix it without changing the reporter source code. You can use our new [onBeforeWrite hook](https://testcafe.io/documentation/404388/guides/advanced-guides/modify-reporter-output?search#article-summary) as follows:

```js
const durationPattern = new RegExp(/<td>(([0-9][0-9]?)h )?(([0-9][0-9]?)m )?(([0-9][0-9]?)s)?<\/td>/g);

module.exports = {
    hooks: {
        reporter: {
            onBeforeWrite: {
                html:  (writeInfo) => {
                    if (writeInfo.initiator !== 'reportTaskDone')
                        return;

                    writeInfo.formattedText = writeInfo.formattedText.replace(durationPattern, (...args) => {
                        const [
                                  _,
                                  hStr = '',
                                  h    = 0,
                                  mStr = '',
                                  m    = 0,
                                  sStr = '',
                                  s    = 0
                              ]        = args
                        const duration = parseInt(s) * 1000 + parseInt(m) * 60 * 1000 + parseInt(h) * 60 * 60 * 1000;

                        return `<td data-value="${duration}">${hStr}${mStr}${sStr}</td>`
                    });
                }
            }
        }
    }
}

```

**Comment by anthony-chu** [comment]
thanks @Artem-Babich for following up and for providing a reference implementation and documentation for the reporter hooks. do you know which version of testcafe supports the new reporter hooks feature?

**Comment by Artem-Babich** [comment]
Hi @anthony-chu,

This hook is available since v2.6.0.
